lib/msal-browser/src/interaction_client/StandardInteractionClient.ts

```diff
@@ -437,12 +439,23 @@ export abstract class StandardInteractionClient extends BaseInteractionClient {
             PerformanceEvents.InitializeBaseRequest,
             this.correlationId
         );
+
+        let configResponseMode = null;
+        if(!this.config.auth.OIDCOptions || 
+            (this.config.auth.OIDCOptions.serverResponseType &&
+            this.config.auth.OIDCOptions.serverResponseType === ServerResponseType.HASH)) {
+            configResponseMode = ResponseMode.FRAGMENT;
```

Comment:
Setting defaults should happen in the configuration builder

Reply:
Can you clarify what you mean by this? The response mode is configured in the OIDCOptions parameter (which is in the configuration builder).

Reply:
If you ensure that `this.config.auth.OIDCOptions` is always defined (either with a default or with the user provided object) then you can reduce all of this (442-451) into just `const configResponseMode = this.config.auth.OIDCOptions.serverResponseType`